v2/tools/generator/internal/astmodel/conversion_function_builder.go

```diff
@@ -397,10 +424,37 @@ func IdentityConvertComplexMapProperty(
 		Body:  astbuilder.StatementBlock(conversion...),
 	}
 
-	result := astbuilder.IfNotNil(
-		params.GetSource(),
-		makeMapStatement,
-		rangeStatement)
+	// If we must forcibly construct empty collections, check if the destination is nil and if so, construct an empty collection
+	// This only applies for top-level collections (we don't forcibly construct nested collections)
+	var result *dst.IfStmt
+	if depth == 0 && builder.ForceEmptyCollections {
+		emptyMap := astbuilder.MakeMap(keyTypeAst, valueTypeAst)
+
+		helpfulComment := &dst.EmptyStmt{
+			Decs: dst.EmptyStmtDecorations{
+				NodeDecs: dst.NodeDecs{
+					End: []string{fmt.Sprintf("// Set property to empty map, as this resource is set to serialize all collections explicitly")},
+				},
+			},
+		}
+
+		result = astbuilder.SimpleIfElse(
+			astbuilder.NotNil(params.GetSource()),
+			[]dst.Stmt{
+				makeMapStatement,
+				rangeStatement,
+			},
+			[]dst.Stmt{
+				helpfulComment,
+				astbuilder.SimpleAssignment(params.GetDestination(), emptyMap),
+			},
+		)
```

Comment:
Applying the above suggestion, it would look something like this:

```suggestion
		assignEmpty := astbuilder.SimpleAssignment(params.GetDestination(), emptyMap)
		astbuilder.AddComments(
			&assignEmpty.Decs.Start,
			"// Set property to empty map, as this resource is set to serialize all collections explicitly")
		assignEmpty.Decs.Before = dst.NewLine
		
		result = astbuilder.SimpleIfElse(
			astbuilder.NotNil(params.GetSource()),
			[]dst.Stmt{
				makeMapStatement,
				rangeStatement,
			},
			[]dst.Stmt{
				helpfulComment,
				assignEmpty,
			},
		)
```

Reply:
Fixed!